pylutron_caseta/cli.py

```diff
@@ -0,0 +1,342 @@
+
+    return functools.update_wrapper(wrapper, main)
+
+
+class _CertOption(click.Option):
+    """An option that accesses a certificate file for the given address."""
+
+    def __init__(self, *args, **kwargs):
+        self.suffix = kwargs.pop("suffix")
+        self.host = kwargs.pop("host")
+        super().__init__(*args, **kwargs)
+
+    def get_default(self, ctx: click.Context, call: bool = True) -> Optional[Path]:
+        if not call:
+            return None
+
+        config_home = xdg.xdg_config_home()
+        base = config_home / "pylutron_caseta"
+
+        config_home.mkdir(exist_ok=True)
+        base.mkdir(mode=0o600, exist_ok=True)
+
+        return base / (self.host(ctx) + self.suffix)
+
+
+class _AddressCertOption(_CertOption):
+    """An option that access a certificate file for a plain hostname."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, host=lambda ctx: ctx.params["address"])
+
+
+class _ResourceCertOption(_CertOption):
+    """An option that access a certificate file for a URL."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args, **kwargs, host=lambda ctx: ctx.params["resource"].hostname
+        )
+
+
+class _UrlParamType(click.ParamType):
+    """A parameter that is urlparsed."""
+
+    name = "url"
+
+    def convert(
+        self, value: str, param: Optional[click.Parameter], ctx: Optional[click.Context]
+    ) -> urllib.parse.ParseResult:
+        if "://" not in value:
+            value = f"leap://{value}"
+        return urlparse(value, "leap", False)
+
+
+URL = _UrlParamType()
+
+
+@click.command()
+@_cli_main
+@click.argument("address")
+@click.option(
+    "--cacert",
+    type=click.File("w", encoding="ascii", atomic=True),
+    cls=_AddressCertOption,
+    suffix="-bridge.crt",
+    help="The path to the CA certificate.",
+)
+@click.option(
+    "--cert",
+    type=click.File("w", encoding="ascii", atomic=True),
+    cls=_AddressCertOption,
+    suffix=".crt",
+    help="The path to the client certificate.",
+)
+@click.option(
+    "--key",
+    type=click.File("w", encoding="ascii", atomic=True),
+    cls=_AddressCertOption,
+    suffix=".key",
+    help="The path to the client certificate key.",
+)
+async def lap_pair(address: str, cacert: TextIO, cert: TextIO, key: TextIO):
+    """
+    Perform LAP pairing.
+
+    This program connects to a Lutron bridge device and initiates the LAP pairing
+    process. The user will be prompted to press a physical button on the bridge, and
+    certificate files will be generated on the local computer.
+
+    By default, the certificate files will be placed in
+    $XDG_CONFIG_HOME/pylutron_caseta, named after the address of the bridge device. The
+    leap tool will look for certificates in the same location.
+    """
+
+    def _ready():
+        click.echo(
+            "Press the small black button on the back of the bridge to complete "
+            + "pairing."
```

Comment:
```suggestion
            "pairing."
```

The + isn't needed (and without it python builds a single string). With the + python builds 2 strings, then adds them building a third).